### PR TITLE
Kotlin String Implementaiton

### LINKED ISSUE #741: [Kotlin] Add strings
# Description

Strings for Kotlin aren't presently on Code Thesaurus. This issue is to add that concept to the system.

# Directions

* On the home page, on the right side (the reference sheet side), choose the language and version, and then the concept.
* The page says the data doesn't exist, but clicking the arrow for the template will provide the template and file for you to use.
* Save that file in `/web/thesauruses/[language]/[version]/strings.json`.
* Fill out all of the `"code"` sections.
* Add in `"not-implemented": "true"` if the concept doesn't exist in the language. You can then delete the `"code"` line.
* Add in `"comment":` if you wish to explain something in the code
* Delete the `"name":` fields before your PR
* Save, commit your code, push to your fork, then make a PR.
* Watch the checks to make sure they all pass. If they don't, click on "Details" and it should tell you what to modify.

# Claiming This Issue

Please leave a comment below so I can assign you to it. This is to prevent multiple people from working on the same issue. 

Check out [CONTRIBUTING.md](https://github.com/codethesaurus/codethesaur.us/blob/main/CONTRIBUTING.md) for more information on how to contribute. 

Check out [https://docs.codethesaur.us/thesaurus/](https://docs.codethesaur.us/thesaurus/) for information on how Code Thesaurus works and how to add to the thesauruses.

# Need Help?

Leave a comment on here with your question, or feel free to ask [@geekygirlsarah](https://twitter.com/geekygirlsarah) on Twitter. You can also email coreteam@codethesaur.us too.

# Thanks!

We really value and appreciate the additions people make to Code Thesaurus, no matter how large or small. So thank you for taking the time to work on this!

### PR DESCRIPTION
**Pull request by apurvabanka**
## What GitHub issue does this PR apply to?

 

Resolves #741 

## What changed and why?

 Implemented changes for Kotlin String.

image

## Checklist

   

- [x] I claimed any associated issue(s) and they are not someone else's
- [x] I have looked at documentation to ensure I made any revisions correctly
- [x] I tested my changes locally to ensure they work
- [x] For language files, I have validated the edited files are valid JSON and data shows up correctly
- [x] For website code edits, I have added or edited any appropriate unit tests for my changes

### CODE REVIEW

**Comment by geekygirlsarah** [comment]
From the validating checks:
```
Error:  `kotlin/1.5/strings.json`, ID: `count_occurrences_of_substring` has not_implemented (underscore) when it should use not-implemented (hyphen)
Error:  `kotlin/1.5/strings.json`, ID: `convert_to_title_case` has not_implemented (underscore) when it should use not-implemented (hyphen)
Error:  `kotlin/1.5/strings.json`, ID: `parameter_format_in_order` has not_implemented (underscore) when it should use not-implemented (hyphen)
3 errors found.
```

I think I also saw a `"not-implemented": "true"`  (with quotes around true) that will need to get fixed. But past that it looks good! Fix those things and I'm happy to merge it in!

(The docker check is an issue related to a library it uses, and isn't a fault of this PR so I'm ignoring it)

**Comment by apurvabanka** [comment]
> From the validating checks:
> 
> ```
> Error:  `kotlin/1.5/strings.json`, ID: `count_occurrences_of_substring` has not_implemented (underscore) when it should use not-implemented (hyphen)
> Error:  `kotlin/1.5/strings.json`, ID: `convert_to_title_case` has not_implemented (underscore) when it should use not-implemented (hyphen)
> Error:  `kotlin/1.5/strings.json`, ID: `parameter_format_in_order` has not_implemented (underscore) when it should use not-implemented (hyphen)
> 3 errors found.
> ```
> 
> I think I also saw a `"not-implemented": "true"` (with quotes around true) that will need to get fixed. But past that it looks good! Fix those things and I'm happy to merge it in!
> 
> (The docker check is an issue related to a library it uses, and isn't a fault of this PR so I'm ignoring it)

My bad !!!! Fixed it.

**Comment by geekygirlsarah** [comment]
> From the validating checks:
> 
> ```
> Error:  `kotlin/1.5/strings.json`, ID: `count_occurrences_of_substring` has not_implemented (underscore) when it should use not-implemented (hyphen)
> Error:  `kotlin/1.5/strings.json`, ID: `convert_to_title_case` has not_implemented (underscore) when it should use not-implemented (hyphen)
> Error:  `kotlin/1.5/strings.json`, ID: `parameter_format_in_order` has not_implemented (underscore) when it should use not-implemented (hyphen)
> 3 errors found.
> ```

Those still need to be fixed, they failed the validating check again.

**Comment by apurvabanka** [comment]
@geekygirlsarah Are there any instructions on how to run the tests?

**Comment by geekygirlsarah** [comment]
I swore we had a docs page on it, but I'm not seeing it. Either I'm tired or it doesn't exist! 😆 

you can run:
`python manage.py validatemetainfofile`
to validate meta_info.json and adding a new language/concept, or 
`python manage.py validatelanginfofiles`
to validate individual files (like what you're doing)

Ideally run both to get a full check on it.

**Comment by geekygirlsarah** [comment]
https://github.com/codethesaurus/docs/issues/42

Clearly I thought this through a year ago! 😆

**Comment by apurvabanka** [comment]
@geekygirlsarah Added some testing docs as well for the contribution page. All test cases are passing now.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -10,6 +10,18 @@ We pledge to act and interact in ways that contribute to an open, welcoming, div
 
 You agree to follow our Code of Conduct while working on the project and/or being a part of any related communities. You can read that [here](https://github.com/codethesaurus/codethesaur.us/blob/main/CODE_OF_CONDUCT.md).
 
+## Testing
```
Comment (geekygirlsarah) [inline CONTRIBUTING.md]: > @geekygirlsarah Added some testing docs as well for the contribution page. All test cases are passing now.

I started having the README point to parts of the documentation for things like this so the README remains simpler and shorter and they can look up whatever they need. Would you mind adding that to do the docs repo instead to go with the issue there, and point the README to that? (Or just undo it and I'll try to add it tomorrow or Monday)

**Review by geekygirlsarah — CHANGES_REQUESTED** [review]
I see the tests pass, so that helps! I'll merge this in with the README adjustments!

File: CONTRIBUTING.md (hunk 1 above)
Comment (apurvabanka) [reply CONTRIBUTING.md]: I'll just remove the readme changes for now.

**Comment by geekygirlsarah** [comment]
Ignoring the Docker build issue, this is good to go! I'll merge it in.

Happy Hacktoberfest! This counts towards that if you're participating!

**Pull request merged**